pkg/operator/target_config_reconciler.go

```diff
@@ -469,83 +525,81 @@ func (c TargetConfigReconciler) sync(ctx context.Context, syncCtx factory.SyncCo
 		return err
 	}
 
-	_, _, err = v1helpers.UpdateStatus(c.ctx, c.kueueClient, func(status *operatorv1.OperatorStatus) error {
-		resourcemerge.SetDeploymentGeneration(&status.Generations, deployment)
-		if deployment != nil {
-			status.ReadyReplicas = deployment.Status.ReadyReplicas
-			status.Conditions = c.buildOperatorConditions(deployment)
+	conditions := c.buildOperatorConditions(deployment)
+	status := applyconfigurationkueueoperatorv1.KueueStatus().WithConditions(conditions...)
```

Comment:
I refactored the status update. Prior to this change, the status update would send a new Timestamp causing an infinite reconcile loop.